hazelcast-client/src/main/java/com/hazelcast/client/cache/impl/ClientCacheProxy.java

```diff
@@ -267,10 +266,10 @@ public void clear() {
         Data keyData = toData(key);
         Data epData = toData(entryProcessor);
 
-        return (T) invokeInternal(keyData, epData, arguments);
+        return (T) toObject(invokeInternal(keyData, epData, arguments));
     }
 
-    protected Object invokeInternal(Data keyData, Data epData, Object... arguments) {
+    protected Data invokeInternal(Data keyData, Data epData, Object... arguments) {
```

Comment:
I moved the `toObject()` to the (only) invoking method, to have the `Data` instance available in the near cached proxy.